torch/_subclasses/fake_tensor.py

```diff
@@ -803,10 +803,18 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
 
         converter = self.fake_tensor_converter
 
-        # If this is a lift, the input tensor is guaranteed to be a
-        # constant, so we keep a copy of the original argument along so
-        # we can query it if we're asked to item() it at some later point
-        if func in self.lift_fns:
+        # To constant propagate through these functions:
+        # 1, If this is a lift, the input tensor is guaranteed to be a
+        #    constant, so we keep a copy of the original argument along so
+        #    we can query it if we're asked to item() it at some later point
+        # 2, Some functions that allow Python numbers to bind to Tensors, e.g, torch.div
+        if func in self.lift_fns or (
+            torch._C._should_allow_numbers_as_tensors(
+                func.name().split("::")[-1].split(".")[0]
+            )
+            and not has_symbolic_sizes
+            and not flat_arg_fake_tensors
+        ):
```

Comment:
I do wonder, though, why constant propagation isn't done by the logic below, at

```
        # We dispatch size/stride/numel on the FakeTensor not its constant, so bail on inplace_view
        all_constant = all(e.constant is not None for e in flat_arg_fake_tensors)
        if (
            torch.Tag.nondeterministic_seeded not in func.tags  # type: ignore[attr-defined]
            and torch.Tag.inplace_view not in func.tags  # type: ignore[attr-defined]
            and all_constant
            and len(flat_arg_fake_tensors) != 0
            and not has_symbolic_sizes
        ):
```

Reply:
`and len(flat_arg_fake_tensors) != 0` condition